source/kraken/tests/apply_disruption_test.cpp

```diff
@@ -109,3 +117,855 @@ BOOST_FIXTURE_TEST_CASE(simple_train_cancellation, SimpleDataset) {
+                              .on(nt::Type_e::StopArea, "stop_area")
+                              .application_periods(btp("20120614T173200"_dt, "20120618T123200"_dt))
+                              .get_disruption(),
+                              *b.data->pt_data, *b.data->meta);
+
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->lines.size(), 2);
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 6); // we have two vj that don't circulate
+
+    bool has_adapted_vj = false;
+    for (const auto* vj: b.data->pt_data->vehicle_journeys) {
+        switch (vj->realtime_level) {
+        case nt::RTLevel::Base:
+            BOOST_CHECK_MESSAGE(ba::ends_with(vj->adapted_validity_pattern()->days.to_string(), "100001"),
+vj->adapted_validity_pattern()->days);
+            BOOST_CHECK_MESSAGE(ba::ends_with(vj->base_validity_pattern()->days.to_string(), "111111"),
+                    vj->base_validity_pattern()->days);
+            break;
+        case nt::RTLevel::Adapted:
+            has_adapted_vj = true;
+            if (vj->uri == "VjA:Adapted:0:stop_area_closed" ||
+                    vj->uri == "VjB:Adapted:0:stop_area_closed") {
+                BOOST_CHECK_MESSAGE(ba::ends_with(vj->adapted_validity_pattern()->days.to_string(), "000000"),
+                        vj->adapted_validity_pattern()->days);
+            } else {
+                BOOST_CHECK_MESSAGE(ba::ends_with(vj->adapted_validity_pattern()->days.to_string(), "011110"),
+                        vj->adapted_validity_pattern()->days);
+            }
+            BOOST_CHECK_MESSAGE(ba::ends_with(vj->base_validity_pattern()->days.to_string(), "000000"),
+                    vj->base_validity_pattern()->days);
+            break;
+
+        case nt::RTLevel::RealTime:
+            //TODO
+            throw navitia::exception("realtime check unhandled case");
+        }
+    }
+    BOOST_REQUIRE(has_adapted_vj);
+
+    b.data->build_uri();
+    b.data->pt_data->index();
+    b.finish();
+    b.data->build_raptor();
+
+    navitia::routing::RAPTOR raptor(*(b.data));
+
+    auto compute = [&](nt::RTLevel level, const std::string& from, const std::string& to,
+            navitia::DateTime datetime, int day) {
+        return raptor.compute(b.data->pt_data->stop_areas_map.at(from), b.data->pt_data->stop_areas_map.at(to),
+                datetime, day, navitia::DateTimeUtils::inf, level, 2_min, true);
+    };
+
+    auto res = compute(nt::RTLevel::Adapted, "A1", "A2", "08:00"_t, 0);
+    BOOST_REQUIRE_EQUAL(res.size(), 1);
+    BOOST_REQUIRE_EQUAL(res[0].items[0].stop_times.size(), 4);
+
+    res = compute(nt::RTLevel::Adapted, "A1", "A2", "08:00"_t, 1);
+    BOOST_REQUIRE_EQUAL(res.size(), 1);
+    BOOST_REQUIRE_EQUAL(res[0].items[0].stop_times.size(), 2);
+    auto vp = res[0].items[0].stop_times[0]->vehicle_journey->adapted_validity_pattern();
+    BOOST_CHECK_MESSAGE(ba::ends_with(vp->days.to_string(), "011110"),  vp->days.to_string());
```

Comment:
minor: I think it would be nice to add test on A1-> stop_area:
like
```
BOOST_REQUIRE(! compute(nt::RTLevel::Base, "A1", "stop_area", "08:00"_t, 1).empty());
BOOST_REQUIRE(compute(nt::RTLevel::Adapted, "A1", "stop_area", "08:00"_t, 1).empty());
```